toolbox

```diff
@@ -30,49 +40,58 @@ toolboxrc="${HOME}"/.toolboxrc
 
 # System defaults
 if [ -f "/etc/default/toolbox" ]; then
-	source "/etc/default/toolbox"
+    source "/etc/default/toolbox"
 fi
 
 # User overrides
 if [ -f "${toolboxrc}" ]; then
-	source "${toolboxrc}"
+    source "${toolboxrc}"
 fi
 
 if [[ -n "${TOOLBOX_DOCKER_IMAGE}" ]] && [[ -n "${TOOLBOX_DOCKER_TAG}" ]]; then
-	TOOLBOX_NAME=${TOOLBOX_DOCKER_IMAGE}-${TOOLBOX_DOCKER_TAG}
+    TOOLBOX_NAME=${TOOLBOX_DOCKER_IMAGE}-${TOOLBOX_DOCKER_TAG}
 else
-	echo "Error: Invalid Docker image name: '${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}'" >&2
-	exit 1
+    echo "Error: Invalid Docker image name: '${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}'" >&2
+    exit 1
 fi
 
 machinename=$(echo "${USER}-${TOOLBOX_NAME}" | sed -r 's/[^a-zA-Z0-9_.-]/_/g')
 machinepath="${TOOLBOX_DIRECTORY}/${machinename}"
 osrelease="${machinepath}/etc/os-release"
+
 if [ ! -f "${osrelease}" ] || systemctl is-failed -q "${machinename}" ; then
-	sudo mkdir -p "${machinepath}"
-	sudo chown "${USER}:" "${machinepath}"
+    sudo mkdir -p "${machinepath}"
+    sudo chown "${USER}:" "${machinepath}"
 
-	# Download and extract the image.
-	sudo --preserve-env docker pull "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}"
-	dcid=$(sudo --preserve-env docker create "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}")
-	sudo --preserve-env docker export -o "${machinepath}/${dcid}.tar" ${dcid}
-	sudo --preserve-env docker rm ${dcid}
-	sudo --preserve-env docker rmi "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}" || true
-	sudo tar xvf "${machinepath}/${dcid}.tar" -C "${machinepath}"
-	sudo rm -f "${machinepath}/${dcid}.tar"
+    if ctr version &> /dev/null; then
+        sudo --preserve-env ctr images pull --platform ${PLATFORM} ${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}
+        sudo ctr images mount --rw --platform ${PLATFORM} ${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG} ${machinepath}
```

Comment:
Will the changes be persistent if just `mount` is used? Otherwise I suggest to keep the current behavior by copying things out.

Reply:
Changement are persistent except if we unmount & pull a newer image

Reply:
Just made some modifications to avoid overwriting, even on reboot. Just tried it and it is ok

Reply:
Thanks! I've had time to try it and it works well.